admin-dev/themes/new-theme/js/app/utils/init-components.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import TranslatableField from '@js/components/translatable-field.js';
+import TranslatableInput from '@js/components/translatable-input.js';
+import TinyMCEEditor from '@js/components/tinymce-editor.js';
+
+const initComponents = () => {
+  let translatableInput = null;
+  let tinyMCEEditor = null;
+  let translatableField = null;
+  // Initialize the prestashop global object
+  window.prestashop.components = {
+    translatableInput: {
+      init() {
+        if (translatableInput === null) {
+          translatableInput = new TranslatableInput();
+        }
+        return translatableInput;
+      },
+    },
```

Comment:
```suggestion
        component: TranslatableInput
    },
```

We discussed about letting the choice to the developer to instanciate the component himself, so maybe we should let the class itself accessible so that we could do:

```js
    new window.prestashop.components.translatableInput.component({localeItemSelector: '.my-selector'});
```

I'm not sure if `component` is the right naming, it's just a suggestion

Reply:
Agreed, it's better to return only the component instead of instantiation.